lib/Mail/DMARC/Base.pm

```diff
@@ -97,13 +98,15 @@ sub any_inet_pton {
         my $file = $self->find_psl_file();
         $public_suffixes_stamp = ( stat( $file ) )[9];
 
-        my $fh = IO::File->new( $file, 'r' )
+        open my $fh, '<:encoding(UTF-8)', $file
             or croak "unable to open $file for read: $!\n";
         # load PSL into hash for fast lookups, esp. for long running daemons
+        my @data = <$fh>;
+        close $fh;
         my %psl = map { $_ => 1 }
                   grep { $_ !~ /^[\/\s]/ } # weed out comments & whitespace
                   map { chomp($_); $_ }    # remove line endings
-                  <$fh>;
+                  @data;
         return $public_suffixes = \%psl;
```

Comment:
Out of curiosity, why read into @data here, vs reading line-by-line from the file handle?